### PR TITLE
Feature/tb 36 switch translate provider to watson

### LINKED ISSUE #36: Switch translate provider to ibm watson
make a wrapper for watson API and refactor translate code to be able to use any translate provider so we can swap them easily, or even provide redundancy

### PR DESCRIPTION
**Pull request by neriat**

### CODE REVIEW

File: teitelbot/config/__init__.py [hunk 1]
```diff
@@ -1,12 +1,20 @@
-import os
+from os import environ
```
Comment (weissgilad) [inline teitelbot/config/__init__.py:1]: Noice

Comment (weissgilad) [reply teitelbot/config/__init__.py:1]: "using os::environ" but ok

File: teitelbot/util/translator/google_translate.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+class GoogleTranslate:
+    def __init__(self):
+        self._translator = googletrans.Translator()
+
+    def translate(self, text, src, dest) -> str:
+        try:
+            return self._translator.translate(text, src=src, dest=dest).text
+        except (KeyError, AttributeError, ValueError) as e:
+            raise RuntimeError("google translate package is unstable. it might got broken again.", e)
```
Comment (weissgilad) [inline teitelbot/util/translator/google_translate.py]: Engrish
```suggestion
            raise RuntimeError("google translate package is unstable and is probably broken again.", e)
```

File: teitelbot/util/translator/ibm_watson_translator.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    def __init__(self):
+        self._translator = LanguageTranslatorV3(
+            version=config.Watson.Translator.VERSION,
+            authenticator=IAMAuthenticator(config.Watson.Translator.API_KEY)
+        )
+        self._translator.set_service_url(config.Watson.Translator.SERVICE_URL)
+
+    def translate(self, text: str, src: str, dest: str) -> str:
+        return (
+            self._translator.translate([text], source=src, target=dest).get_result()['translations'][0]['translation']
```
Comment (weissgilad) [inline teitelbot/util/translator/ibm_watson_translator.py]: What does this syntax mean?

File: teitelbot/config/__init__.py (hunk 1 above)
Comment (neriat) [reply teitelbot/config/__init__.py:1]: One is human readable, other is machine code

File: teitelbot/util/translator/ibm_watson_translator.py (hunk 3 above)
Comment (neriat) [reply teitelbot/util/translator/ibm_watson_translator.py]: Do you mean `['translations'][0]['translation']`?
that's parsing of the returned json

File: teitelbot/config/__init__.py (hunk 1 above)
Comment (weissgilad) [reply teitelbot/config/__init__.py:1]: One is beautiful and human readable, the other is python

File: teitelbot/util/translator/ibm_watson_translator.py (hunk 3 above)
Comment (weissgilad) [reply teitelbot/util/translator/ibm_watson_translator.py]: Would it be better to separate into variables?

File: teitelbot/util/translator/google_translate.py (hunk 2 above)
Comment (neriat) [reply teitelbot/util/translator/google_translate.py]: done, (-_-)

**Review by weissgilad — APPROVED** [review]

**Pull request merged**